### PR TITLE
Analyze Crashes

### PR DESCRIPTION
**Pull request by mjohnson541**
This provides functionality for analyzing why a mechanism simulation crashed in terms of reactions, kinetics and thermochemistry. Also adds some functionality for analyzing collision limit violations, but only relative to H+H->H2 since sigma and epsilon often aren't available.

### CODE REVIEW

File: src/Debugging.jl [hunk 1]
```diff
@@ -1,2 +1,30 @@
...
+
+@with_kw struct DebugReaction
+    reactants::Array{DebugSpecies,1}
+    products::Array{DebugSpecies,1}
+    rxnstring::String
+    tol::Float64 = 0.0
+    kf::Float64 = 0.0
+    krev::Float64 = 0.0
+    rt::Float64 = 0.0
+    ratio::Float64 = 0.0
```
[the remark this answers is not in the document]
Comment (hwpang) [reply src/Debugging.jl:21]: From developer hour: rt is rate, ratio is attempt to scale the rate so we have a way to threshold the tolerance, which is rt/medium of rates

**Review by hwpang — CHANGES_REQUESTED** [review]
Mostly looks good to me! Only a couple of requests. Could you also write a jupyter notebook using these functions? It's just easier to point new users to a jupyter notebook to learn how to use new stuff

File: src/Debugging.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjohnson541) [reply src/Debugging.jl:33]: added docstrings

**Comment by mjohnson541** [comment]
I think I've made all the requested changes.

**Review by hwpang — APPROVED** [review]
Looks good to me! I will merge it.

**Pull request merged**